libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslDiagnostics.java

```diff
@@ -415,23 +440,40 @@ private static boolean isSelfIssued(X509Certificate certificate) {
     }
 
     private static String keyUsageDescription(X509Certificate certificate) {
-        return Optional.ofNullable(certificate.getKeyUsage())
-            .map(keyUsage -> "keyUsage [" + Arrays.toString(keyUsage) + "]")
+        boolean[] keyUsage = certificate.getKeyUsage();
+        if (keyUsage == null || keyUsage.length == 0) {
+            return "no keyUsage";
+        }
+
+        final String[] keyUsageGlossary = {"digitalSignature", "nonRepudiation", "keyEncipherment",
```

Comment:
Should I use an enum here too like I've done with `ExtendedKeyUsage`, for uniformity? I could define the index as the Enum field.